### ISSUE TITLE
Update data silent failure

### ISSUE THREAD

**Issue by iendjinn** [nqminds/nqm-api-tdx#20]
If many calls are made to updateData in rapid succession (not simultaneously) data will intermittently fail to be written to the dataset. No error or warning is produced.

Minimal reproduction: https://github.com/IGlautier/nqm-update-data-silent-failure

**Comment by TobyEalden** [comment]
Thanks for this. Do you have the share token details and I'll take a look?

**Comment by iendjinn** [comment]
The dataset is public view and edit but you can use:

```
const shareKeyId = "HJgaMhc4zH";
const shareKeySecret = "throwaway";
```

**Comment by TobyEalden** [comment]
Fixed - this was a bug in the core rather than the client API, but fixed in [v0.5.0](https://github.com/nqminds/nqm-tdx/releases/tag/v0.5.0)
